rktl_sim/nodes/simulator

```diff
@@ -160,60 +122,70 @@ class Simulator(object):
                 pass
 
     def check_urdf(self, param):
-        """Validates that a URDF exists, then returns path"""
+        """
+        Validates that a URDF exists, then returns its path.
+        @param param: The path to the URDF.
+        """
+
         if param is None:
             rospy.signal_shutdown('no urdf path set for "{}"'.format(param))
 
         i = 0
         while not os.path.isfile(param) and i < 5:
-            rospy.sleep(0.1) # Wait for xacro build
+            # Wait for xacro build.
+            rospy.sleep(0.1)
             i += 1
-        
+
         if not os.path.isfile(param):
             rospy.signal_shutdown(
                 'no urdf file exists at path {}'.format(param))
 
     def effort_cb(self, effort_msg, car_id):
-        """Sets a car's effort (overwrites commands)"""
+        """Sets a car's effort (overwrites commands)."""
         self.cmd_lock.acquire()
         self.sim.set_car_command(car_id,
-            (effort_msg.throttle, effort_msg.steering))
+                                 (effort_msg.throttle, effort_msg.steering))
         self.cmd_lock.release()
 
     def cmd_cb(self, cmd_msg, car_id):
-        """Sets a car's command (overwrites efforts)"""
+        """Sets a car's command with the velocity and curvature."""
         self.cmd_lock.acquire()
         self.sim.set_car_command(car_id,
-            (cmd_msg.velocity, cmd_msg.curvature))
+                                 (cmd_msg.velocity, cmd_msg.curvature))
         self.cmd_lock.release()
 
     def create_car_cb(self, req):
-        """Adds a new car to the simulator"""
+        """Add new car to the simulator."""
         self.reset_lock.acquire()
         if req.name in self.car_ids:
+            self.reset_lock.release()
             return CreateCarResponse(False)
 
         init_pose = None
+        # Checks if resetting to random position.
+        # if so, the sim.create_car will handle the random generation.
         if req.reset_to_random_pose is False:
             init_pose = {
                 "pos": [req.init_pos.x,
-                        req.init_pos.y, 
+                        req.init_pos.y,
                         req.init_pos.z],
                 "orient": [0.0, 0.0, req.init_yaw],
             }
 
         self.car_ids[req.name] = self.sim.create_car(
-            'car', init_pose, noise=self.car_noise, props=self.car_properties) 
+            'car', init_pose, noise=self.car_noise, props=self.car_properties)
 
+        # Subscribers
         car_id = self.car_ids[req.name]
         self.car_effort_subs[req.name] = rospy.Subscriber(
             f'/cars/{req.name}/effort', ControlEffort,
             self.effort_cb, callback_args=car_id)
-        self.car_cmd_subs[req.name] =  rospy.Subscriber(
+        self.car_cmd_subs[req.name] = rospy.Subscriber(
             f'/cars/{req.name}/command', ControlCommand,
             self.cmd_cb, callback_args=car_id)
```

Comment:
So we still want to change this function to do the set param implementation. `sim.create_car` should only be called in our 'global' reset function.